llvm/lib/CodeGen/MachineFunction.cpp

```diff
@@ -383,13 +383,28 @@ void MachineFunction::RenumberBlocks(MachineBasicBlock *MBB) {
 /// and the SectionID's are assigned to MBBs.
 void MachineFunction::assignBeginEndSections() {
   front().setIsBeginSection();
+  front().setIsFirstNonEmptyBBInSection();
```

Comment:
Is this line intentional? I thought you want to check before setting this flag (i.e. line 392).

Reply:
Good catch, thanks! I think I was originally under the assumption that the first block has to be non-empty. But you are right, this should be removed.